From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BillyGalbreath <Blake.Galbreath@GMail.com>
Date: Sun, 7 Oct 2018 00:54:15 -0500
Subject: [PATCH] Add sun related API


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 94277942f5779cfc8ad7ab5d27f43d891da1297c..313602704694c85bbac96b68deab67cd25f9913c 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -1760,6 +1760,16 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
      */
     public void setFullTime(long time);
 
+    // Paper start
+
+    /**
+     * Check if it is currently daytime in this world
+     *
+     * @return True if it is daytime
+     */
+    public boolean isDayTime();
+    // Paper end
+
     /**
      * Gets the full in-game time on this world since the world generation
      *
diff --git a/src/main/java/org/bukkit/entity/Mob.java b/src/main/java/org/bukkit/entity/Mob.java
index af5158365e96e0934974836a4e2915dd9f161ac5..141479aae73666f1c1cb4ac95931c171fd8d4407 100644
--- a/src/main/java/org/bukkit/entity/Mob.java
+++ b/src/main/java/org/bukkit/entity/Mob.java
@@ -21,6 +21,13 @@ public interface Mob extends LivingEntity, Lootable {
      */
     @NotNull
     com.destroystokyo.paper.entity.Pathfinder getPathfinder();
+
+    /**
+     * Check if this mob is exposed to daylight
+     *
+     * @return True if mob is exposed to daylight
+     */
+    boolean isInDaylight();
     // Paper end
     /**
      * Instructs this Mob to set the specified LivingEntity as its target.
