From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <blake.galbreath@gmail.com>
Date: Thu, 14 Oct 2021 12:36:46 -0500
Subject: [PATCH] Left handed API


diff --git a/src/main/java/org/bukkit/entity/Mob.java b/src/main/java/org/bukkit/entity/Mob.java
index eca7a38cb7c08e0f19f9a31f5efde2a52381d336..09794c42537cbf7c4b0b28666d3df0b5342adae6 100644
--- a/src/main/java/org/bukkit/entity/Mob.java
+++ b/src/main/java/org/bukkit/entity/Mob.java
@@ -205,4 +205,22 @@ public interface Mob extends LivingEntity, Lootable {
      */
     void setAggressive(boolean aggressive);
     // Paper end - Missing Entity API
+
+    // Paper end - left-handed API
+
+    // Paper start
+    /**
+     * Check if Mob is left-handed
+     *
+     * @return True if left-handed
+     */
+    public boolean isLeftHanded();
+
+    /**
+      * Set if Mob is left-handed
+      *
+      * @param leftHanded True if left-handed
+      */
+    public void setLeftHanded(boolean leftHanded);
+    // Paper end - left-handed API
 }
