From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nassim Jahnke <nassim@njahnke.dev>
Date: Sun, 24 Oct 2021 15:49:35 +0200
Subject: [PATCH] Fix Bukkit NamespacedKey shenanigans


diff --git a/src/main/java/com/destroystokyo/paper/loottable/PaperLootableBlock.java b/src/main/java/com/destroystokyo/paper/loottable/PaperLootableBlock.java
index 132c7d256dfb3d18be2955c87ab034dcddf4c7d0..61bd379668895b50eea5b6067cf11c3b7971d2cd 100644
--- a/src/main/java/com/destroystokyo/paper/loottable/PaperLootableBlock.java
+++ b/src/main/java/com/destroystokyo/paper/loottable/PaperLootableBlock.java
@@ -19,6 +19,7 @@ public interface PaperLootableBlock extends PaperLootable {
         }
 
         final ResourceLocation key = this.getRandomizableContainer().getLootTable();
+        if (key.getPath().isEmpty()) return null; // Paper - fix some NamespacedKey parsing
         return Bukkit.getLootTable(CraftNamespacedKey.fromMinecraft(key));
     }
 
diff --git a/src/main/java/com/destroystokyo/paper/loottable/PaperLootableEntity.java b/src/main/java/com/destroystokyo/paper/loottable/PaperLootableEntity.java
index 0d7bb7b23a84351d75060089b6e0037ebfed3b75..e7ef9a2c5e58d4a285c5f519f5a02a35a7adf591 100644
--- a/src/main/java/com/destroystokyo/paper/loottable/PaperLootableEntity.java
+++ b/src/main/java/com/destroystokyo/paper/loottable/PaperLootableEntity.java
@@ -20,6 +20,7 @@ public interface PaperLootableEntity extends Lootable {
         }
 
         final ResourceLocation key = this.getHandle().getLootTable();
+        if (key.getPath().isEmpty()) return null; // Paper - fix some NamespacedKey parsing
         return Bukkit.getLootTable(CraftNamespacedKey.fromMinecraft(key));
     }
 
diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftNamespacedKey.java b/src/main/java/org/bukkit/craftbukkit/util/CraftNamespacedKey.java
index 5f40d240b879e3989897b6e45725a8e5a6a7f194..5014192edb9616ce725fc1592832034789527b6f 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftNamespacedKey.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftNamespacedKey.java
@@ -13,7 +13,7 @@ public final class CraftNamespacedKey {
             return null;
         }
         ResourceLocation minecraft = ResourceLocation.tryParse(string);
-        return (minecraft == null) ? null : CraftNamespacedKey.fromMinecraft(minecraft);
+        return (minecraft == null || minecraft.getPath().isEmpty()) ? null : CraftNamespacedKey.fromMinecraft(minecraft); // Paper - Bukkit's parser does not match Vanilla for empty paths
     }
 
     public static NamespacedKey fromString(String string) {
